src/orion/core/worker/consumer.py

```diff
@@ -52,9 +53,15 @@ def __init__(self, experiment):
         self.template_builder = SpaceBuilder()
         self.template_builder.build_from(experiment.metadata['user_args'])
         # Get path to user's script and infer trial configuration directory
+
         self.script_path = experiment.metadata['user_script']
-        self.tmp_dir = os.path.join(tempfile.gettempdir(), 'orion')
-        os.makedirs(self.tmp_dir, exist_ok=True)
+
+        if experiment.working_dir:
+            self.dir = os.path.abspath(experiment.working_dir)
+        else:
+            self.dir = os.path.join(tempfile.gettempdir(), 'orion')
+
+        os.makedirs(self.dir, exist_ok=True)
```

Comment:
You can push this inside `WorkingDir`.

Reply:
It'll get call once instead of being call every time the `WorkingDir` is created, which is every time `consume` is called. So, while it'll be call multiple times for nothing, because this doesn't change throughout execution.

Reply:
Agreed, but os.makedirs is called every time for the subfolder anyway. Also, this is optimizing milliseconds in favor of code simplicity, while there is bottlenecks that are significantly greater here (most particularly the HP sampling).

Reply:
Alright, I'll move it.

Reply:
Actually, you can simply **re**move it. :upside_down_face: